swarm/fuse/swarmfs_unix.go

```diff
@@ -159,28 +161,59 @@ func (swarmfs *SwarmFS) Mount(mhash, mountpoint string) (*MountInfo, error) {
 	go func() {
 		log.Info("swarmfs", "serving hash", mhash, "at", cleanedMountPoint)
 		filesys := &SwarmRoot{root: rootDir}
+		//start serving the actual file system; see note below
 		if err := fs.Serve(fconn, filesys); err != nil {
 			log.Warn("swarmfs could not serve the requested hash", "error", err)
 			serverr <- err
 		}
-
+		mi.serveClose <- struct{}{}
```

Comment:
This is ok only if there is one receiver for serveClose, which is the case here. If there are other places in the code that need to block until receiving from serveClose, then this needs to change to close(mi.serveClose).

Reply:
took the freedom to leave it as-is as there is only one place and probably will always be so, but thanks for pointing that out

Reply:
Agreed.